### PR TITLE
updated filter functionality on my projects page

### PR DESCRIPTION
**Pull request by russbiggs**
Adds osm_id filtering based on user.json array

### CODE REVIEW

**Comment by localjo** [comment]
I'm getting an error on `/user/{user_id}/campaigns`:
```
File "/flask_project/campaign_manager/utilities.py", line 304, in get_data_from_s3
campaign_json['geojson'] = geojson
TypeError: list indices must be integers or slices, not str
```

File: flask_project/campaign_manager/templates/campaign_index.html [hunk 1]
```diff
@@ -414,20 +414,24 @@ <h2>Create your first project</h2>
...
+    var osm_id = pageURL.substr(pageURL.lastIndexOf('/') + 1);
     $.ajax({
-      url: "/campaigns",
+      url: `/user/${osm_id}/campaigns`,
       success: function (data) {
         hideLoading();
-        if (data.length > 1) {
+        if (data.length > 0) {
           data.map(function (campaign) {
+            console.log(campaign)
```
Comment (localjo) [inline flask_project/campaign_manager/templates/campaign_index.html]: Did you mean to leave this `console.log` line in, and the one on line 434?

**Comment by localjo** [comment]
The error I was getting was because I didn't have `{user_id}.json` set up correctly with my projects in it. I set that file up correctly, and now this works for me.

File: flask_project/campaign_manager/templates/campaign_index.html (hunk 1 above)
Comment (russbiggs) [reply flask_project/campaign_manager/templates/campaign_index.html]: nope, removing now

**Pull request merged**